File: 06/code_from_slides/regex.cpp
#include <iostream>
#include <regex>

int main() {
    std::string input;
    std::regex integer{"-?\\d+"};
    std::cout << "Enter some integers (Ctrl-d to exit):" << std::endl;

    while(std::cin>>input) {
        if(regex_match(input,integer)) std::cout << "That’s an int!" << std::endl;
        else std::cout << "***INVALID INPUT***" << std::endl;
    }
}

File: 06/java/Regex.java
import java.util.Scanner;
import java.util.NoSuchElementException;

class Regex {
    public static void main(String[] args) {
        String pattern = "-?\\d+";
        System.out.println("Enter some integers:");
        
        Scanner in = new Scanner(System.in);
        String line = null;

        while(true) {
            try {
                line = in.next(pattern);
                System.out.println("That's an int!");
            } catch (NoSuchElementException e) {
                if(!in.hasNext()) break;
                System.err.println("***INVALID INPUT*** ");
                line = in.nextLine();
            }
        }
    }
}
        
